whelk-core/src/main/groovy/whelk/Whelk.groovy

```diff
@@ -187,8 +188,8 @@ class Whelk implements Storage {
             }
         }
 
-        // If the number of dependers isn't too large. Update them synchronously
-        if (dependers.size() < 20) {
+        // If the number of dependers isn't too large or we are inside a batch job. Update them synchronously
+        if (dependers.size() < 20 || (batchJobThread() && indexers.activeCount() > 30) ) {
```

Comment:
Using `indexers.activeCount()` is a bit sketchy...
The idea is to not let the number of reindexer threads grow unbounded if we are inside a slow batch job